presto-spi/src/main/java/com/facebook/presto/spi/statistics/HistoryBasedPlanStatisticsProvider.java

```diff
@@ -36,6 +40,15 @@
      */
     Map<PlanNodeWithHash, HistoricalPlanStatistics> getStats(List<PlanNodeWithHash> planNodesWithHash);
 
+    /**
+     * Overloaded method to use by external providers
+     */
+    default Map<PlanNodeWithHash, HistoricalPlanStatistics> getStats(List<PlanNodeWithHash> planNodesWithHash,
```

Comment:
Because this function is in presto-spi, we need to clean up its arguments.  Lets aim for:

`Map<PlanNodeWithHash, HistoricalPlanStatistics> getStats(ConnectorSession session, List<PlanNodeWithHash> planNodesWithHash)`

You can modify and rename PlanNodeWithHash to be an interface where we can get input table stats, plan node, jsonText() etc.